File: typescript/difference-of-squares/difference-of-squares.ts
import { totalmem } from "os";
import { numberLiteralTypeAnnotation } from "@babel/types";

class Squares {
    squareOfSum: number;
    sumOfSquares: number;
    difference: number;

    constructor(size: number) {
        let numbers = Array.from(Array(size + 1).keys());
        this.sumOfSquares = numbers.reduce((total: number, value: number): number => total += (value ** 2));
        this.squareOfSum = numbers.reduce((total: number, value: number): number => total += value) ** 2;
        this.difference = this.squareOfSum - this.sumOfSquares;
    }

}

export default Squares

File: dart/difference-of-squares/lib/difference_of_squares.dart
class DifferenceOfSquares {

  int sumOfSquares(int aNumber) {
    var values = List.generate(aNumber + 1, (i) => i);
    var sum = values.reduce((total, value) => total + value * value);

    return sum;
  }

  int differenceOfSquares(int aNumber) {
    return squareOfSum(aNumber) - sumOfSquares(aNumber);

  }

  int squareOfSum(int aNumber) {
    var values = List.generate(aNumber + 1, (i) => i);
    var sum = values.reduce((total, value) => total + value);

    return sum * sum;
  }
}
